### ISSUE TITLE
HexagonLayer elevation based on getColorValue

### ISSUE THREAD

**Issue by MarshallChen** [uber/deck.gl#674]
![image](https://cloud.githubusercontent.com/assets/1115805/26665785/9105f340-46ce-11e7-9774-f6f197e3df2b.png)

This is the map what I got, I have a customized `getColorValue` on props, how can I make the hexbins elevation based on this value? Seems currently the elevation is based on `points.length` and I can't customize it?
```js
_onGetSublayerElevation(cell) {
    ...
    const domain = elevationDomain || [0, sortedBins.maxCount];
    return linearScale(domain, elevationRange, cell.points.length);
}
```
Maybe change to this?
```js
const { getColorValue } = this.props;
...
return linearScale(domain, elevationRange, getColorValue(cell.points));
```

**Comment by shaojingli** [comment]
@heshan0131 what do you think?

**Comment by heshan0131** [comment]
I agree that elevation can be based on other metrics besides count. However I would not have it based on the same value that color is based on. 

Color and Elevation are two different visual channels that can be used to encode different metrics.  E.g. as it is implemented right now. With a heatmap showing population demographic, elevation can be used to show population density and color can be used to show average household income. In which case, we can tell the story of how income correlates to density.

That being said. I think we can add another props `getElevationValue` similar as `getColorValue`, it would have the binning and sorting run twice on ever data change. Because it needs to sort the hex bins by values that color is based on AND values that elevation is based on. User can always pass the same function if they want the elevation to correlate with the color.

**Comment by shaojingli** [comment]
Then let's use this issue to track this task - adding `getElevationValue` to hex/grid layers

**Comment by shaojingli** [comment]
@heshan0131 is this done in your latest changes?

**Comment by heshan0131** [comment]
Not yet. it's coming alone

**Comment by 1chandu** [comment]
@heshan0131  It is resolved by #938, can we close it?
